### DISCUSSION TITLE
SOLVED!!! Vulnerabilities detection work on Linux agent but not on windows agent

### DISCUSSION THREAD

**Discussion by tismofied** [wazuh/wazuh#25014] [Q&A] [answered]
in my humble homelab where I tinker with proxmox and docker containers, I installed an ubuntu 22.04 LXC from proxmox helperscripts and installed wazuh on it.
everything seems to be returning green checkmarks except for the fact windows 11 vulnerability detector or any detector for that matter is not loading at all. I've attached images showing detections in proxmox endpoint are working.

``` 
<scan>
    <enabled>yes</enabled>
    <scan_on_start>yes</scan_on_start>
    <interval>12h</interval>
    <skip_nfs>yes</skip_nfs>
  </scan>

  <vulnerability-detection>
    <enabled>yes</enabled>
    <index-status>yes</index-status>
    <feed-update-interval>60m</feed-update-interval>
  </vulnerability-detection>

  <indexer>
    <enabled>yes</enabled>
    <hosts>
      <host>https://127.0.0.1:9200</host>
    </hosts>
    <ssl>
      <certificate_authorities>
        <ca>/etc/filebeat/certs/root-ca.pem</ca>
      </certificate_authorities>
      <certificate>/etc/filebeat/certs/wazuh-server.pem</certificate>
      <key>/etc/filebeat/certs/wazuh-server-key.pem</key>
    </ssl>
  </indexer>
```

detections are enabled in ossec.conf 

![2024-08-04 15 19 49 192 168 10 157 433f92bc377f](https://github.com/user-attachments/assets/56dd0243-7838-4c13-8016-f9bbc6e04fc7)
![2024-08-04 15 20 20 192 168 10 157 200f88f07590](https://github.com/user-attachments/assets/6400310d-b699-4f5e-bdd6-a33872f29320)
![2024-08-04 15 20 53 192 168 10 157 2b376799b172](https://github.com/user-attachments/assets/909543a2-c568-4c11-8063-32e24171d1f3)
![2024-08-04 15 23 46 192 168 10 157 3201a1b64e67](https://github.com/user-attachments/assets/7f91e281-84f9-47c1-9a52-7e6f7e545080)

can someone help with might be the issue?

**Comment by Damian-Mangold** [comment]
Hi @tismofied, in order to analyze the issue and perform some tests, could you please provide me with the following information from the agents where you're observing the issue?

- [OS information](https://documentation.wazuh.com/current/user-manual/api/reference.html#operation/api.controllers.syscollector_controller.get_os_info)
- [Package inventory](https://documentation.wazuh.com/current/user-manual/api/reference.html#operation/api.controllers.syscollector_controller.get_packages_info)
- Vulnerabilities inventory
- [Hotfix information](https://documentation.wazuh.com/current/user-manual/api/reference.html#operation/api.controllers.syscollector_controller.get_hotfix_info) (for Windows OS)

Thank you!

**Reply by tismofied** [reply]
.

**Reply by Damian-Mangold** [reply]
Hi @tismofied, let me analyse the information you shared with me and I will get back to you as soon as possible.

**Comment by Damian-Mangold** [comment]
Hi @tismofied, after reviewing the information you provided, everything appears to be in order. The package information and hotfix data are being correctly collected from the Windows agent. The absence of identified vulnerabilities does not necessarily indicate that VD is not functioning; it could simply mean that your OS is up to date with the latest version and the installed packages do not have known vulnerabilities.

As a double check, please run the following command to check for any error messages in the log file:

- `cat /var/ossec/logs/ossec.log | grep -E "ERROR|WARN"`

This will help ensure that there are no underlying issues.

**Reply by tismofied** [reply]
> * cat /var/ossec/logs/ossec.log | grep -E "ERROR|WARN"

Hi @Damian-Mangold , thank you for getting back to me swiftly. 
I remember running that cmd before and I did run it just now on the host that's running wazuh but nothing returns. no outout whatsoever.

**Reply by Damian-Mangold** [reply]
Ok, so everything seems to be in order. If you detect any anomalous behaviour, or identify any false negatives, please inform me and I will check it in more detail.

**Comment by tismofied** [comment, marked as answer]
SOLVED!
in case anyone else encounter the same issue, I solved it by setting up indexer username and password per Wazuh documentation
```
/var/ossec/bin/wazuh-keystore -f indexer -k username -v <INDEXER_USERNAME>

/var/ossec/bin/wazuh-keystore -f indexer -k password -v <INDEXER_PASSWORD>
```
then going to dashboard management > index patterns > wazuh-state-vulnerabilties > refresh field list
